File: Minimum Absolute Distance Between Mirror Pairs/Minimum Absolute Distance Between Mirror Pairs.java
// Solution for Minimum Absolute Distance Between Mirror Pairs in JAVA

class Solution {
    public int minMirrorPairDistance(int[] nums) {
        Map<Integer, Integer> reverseIndexMap = new HashMap<>();
        int minimumDistance = Integer.MAX_VALUE;

        for (int currentIndex = 0; currentIndex < nums.length; currentIndex++) {
            int currentValue = nums[currentIndex];

            // Check if current value matches any previously stored reversed value
            if (reverseIndexMap.containsKey(currentValue)) {
                int previousIndex = reverseIndexMap.get(currentValue);
                minimumDistance = Math.min(minimumDistance, currentIndex - previousIndex);
            }

            // Store reversed value
            int reversedValue = getReversedNumber(currentValue);
            reverseIndexMap.put(reversedValue, currentIndex);
        }

        return (minimumDistance == Integer.MAX_VALUE) ? -1 : minimumDistance;
    }

    private int getReversedNumber(int number) {
        StringBuilder sb = new StringBuilder(String.valueOf(number));
        sb.reverse();
        return Integer.parseInt(sb.toString()); // removes leading zeros
    }
}

File: Minimum Absolute Distance Between Mirror Pairs/Minimum Absolute Distance Between Mirror Pairs.py
# Solution for Minimum Absolute Distance Between Mirror Pairs in PY

class Solution:
    def minMirrorPairDistance(self, nums):
        reverse_index_map = {}
        minimum_distance = float('inf')

        def get_reversed_number(number):
            return int(str(number)[::-1])  # removes leading zeros automatically

        for current_index, current_value in enumerate(nums):

            # Check if current value matches any stored reversed value
            if current_value in reverse_index_map:
                previous_index = reverse_index_map[current_value]
                minimum_distance = min(minimum_distance, current_index - previous_index)

            # Store reversed value
            reversed_value = get_reversed_number(current_value)
            reverse_index_map[reversed_value] = current_index

        return -1 if minimum_distance == float('inf') else minimum_distance
